src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -127,8 +127,8 @@ public static SourceText From(Stream stream, Encoding encoding = null, SourceHas
                 throw new InvalidDataException();
             }
 
-            var checksum = CalculateChecksum(stream, checksumAlgorithm);
-            return new StringText(text, encoding, checksum, checksumAlgorithm);
+            Debug.Assert(encoding != null);
+            return new StringText(text, encoding, checksumAlgorithm: checksumAlgorithm);
```

Comment:
That is not correct: "the checksum should be calculated with the decoded string rather than the original bytes since the decoded string is the text represented "

Checksum should always be calculated from the original bytes. That's how the debugger calculates it. If we do it differently it will break debugging.